CMSIS: update to CMSIS 5.7.0

0xc0170 opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html).

NOTE: Do not remove any of the template headings (even for optional sections) as this
template is automatically parsed. 
-->

### Summary of changes <!-- Required -->

<!-- 
    Please provide the following information: 

    Description of the the change (what is this fixing / adding / removing?).

    Why the change is needed (if this is fixing a reported issue please summarize what
    the issue is and add the reference. E.g. Fixes #17119).

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
    
-->

Fixes #12568

There are couple of fixes here:

- importer be compatible with python 3
- importer sha for RTOS after move (sha updated , after I manually resolved the conflicts)

#### Impact of changes <!-- Optional -->
<!-- 
    If there are any implications for users taking this change then they must be 
    provided here. For Major PR types this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

#### Migration actions required <!-- Optional -->
<!-- 
    This should only be applicable in Major PR types for which this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

### Documentation <!-- Required -->

<!-- 
    Please provide details of any document updates required, including links to any
    related PRs against the docs repository.
    If no document updates are required please specify 'None', this at least tells us
    that this has been considered.
-->

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front of them would change the meaning incorrectly. 
-->
    [X] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Provide all the information required, listing all the testing performed. For new targets please attach full test results for all supported compilers.
-->
    [] No Tests required for this change (E.g docs only update)
    [X] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

<!--
    Request additional reviewers with @username or @team
-->

----------------------------------------------------------------------------------------------------------------

bulislaw committed:
CMSIS/RTX: Patch RTX4 to preserve osThreadDef compatibility

mbed OS used older RTX4 version and with osThreadDef accepting only 3
parameters, to preserve compatibility we hardcode the 'instances'
parameter to 1.

(cherry picked from commit 428acae1b2ac15c3ad523e8d40755a9301220822)
(cherry picked from commit 4360b7bbf815c4d812005938c9c27af199803a97)

bulislaw committed:
CMSIS/RTX: Patch RTX4 to preserve osThreadDef compatibility

mbed OS used older RTX4 version and with osThreadDef accepting only 3
parameters, to preserve compatibility we hardcode the 'instances'
parameter to 1.

(cherry picked from commit 428acae1b2ac15c3ad523e8d40755a9301220822)
(cherry picked from commit 4360b7bbf815c4d812005938c9c27af199803a97)

bulislaw committed:
CMSIS/RTX: Pre-processor defines used for assembly

CMSIS repo does not support pre-processor defines, hence multiple assembly
files are added for secure/non-secure and floating point tools.
Mbed OS tools support assembly file pre-processing, but the build system
does not support multiple assembly files for each target, hence updating
the assembly files.

1. Patch RTX so irq_cm4f.S files work with no FPU targets
2. Patch RTX so irq_armv8mml.S files to work with and without FPU
2. Patch RTX so irq_armv8mml.S and irq_armv8mbl.S files to work with secure and
   non-secure builds

(cherry picked from commit 96e0689204d375e23bf69d7787a18ba07182f085)

bulislaw committed:
CMSIS/RTX: Pre-processor defines used for assembly

CMSIS repo does not support pre-processor defines, hence multiple assembly
files are added for secure/non-secure and floating point tools.
Mbed OS tools support assembly file pre-processing, but the build system
does not support multiple assembly files for each target, hence updating
the assembly files.

1. Patch RTX so irq_cm4f.S files work with no FPU targets
2. Patch RTX so irq_armv8mml.S files to work with and without FPU
2. Patch RTX so irq_armv8mml.S and irq_armv8mbl.S files to work with secure and
   non-secure builds

bulislaw committed:
CMSIS/RTX: Allow overwriting mutex ops for ARMC

bulislaw committed:
CMSIS/RTX: Allow overwriting mutex ops for ARMC

(cherry picked from commit 08ab8cc47d8722bf0c767990cd615cf1c427d006)

Commit:
CMSIS/RTX: Patch to conditionally compile

tz_context.c should be compiled only for secure world,
definition of API's in tz_context.h should be part of secure
binary/bootloader when building mbed-os as non-secure

(Cherry picked from d0a43b8af0eef4775ec4c3da5994ecceb9ed4558)

(cherry picked from commit fb354752eb69403ad503c8e53da67da6483776d6)

Commit:
CMSIS/RTX: Patch to conditionally compile

tz_context.c should be compiled only for secure world,
definition of API's in tz_context.h should be part of secure
binary/bootloader when building mbed-os as non-secure

(Cherry picked from d0a43b8af0eef4775ec4c3da5994ecceb9ed4558)

(cherry picked from commit fb354752eb69403ad503c8e53da67da6483776d6)

bulislaw committed:
CMSIS/RTX: Fix using FALSE/TRUE with preprocesor

bulislaw committed:
CMSIS/RTX: Fix using FALSE/TRUE with preprocesor

(cherry picked from commit dd21ea0ae0559f148d3ff5b1a1937f9d7e0e1138)

bulislaw committed:
CMSIS/RTX: Reintroduce arm_math.h

(cherry picked from commit 6a6e3ac0ebab1a6b6aa08d0928702c79562acee9)

bulislaw committed:
CMSIS/RTX: Reintroduce arm_math.h

(cherry picked from commit 6a6e3ac0ebab1a6b6aa08d0928702c79562acee9)

Commit:
CMSIS/RTX: Move Idle and Timer thread stack to bss section.

In case of ARM compiler, idle and timer thread stack though assigned
to `.bss.os` section since not zero initialized are part of `data` section.

In this commit, we are moving stacks of idle and timer thread to bss
section and thereby saving ROM space.

(cherry picked from commit 9549fff786475bdcd6ab1d8ac8db1c8618c19f6f)

Commit:
CMSIS/RTX: Move Idle and Timer thread stack to bss section.

In case of ARM compiler, idle and timer thread stack though assigned
to `.bss.os` section since not zero initialized are part of `data` section.

In this commit, we are moving stacks of idle and timer thread to bss
section and thereby saving ROM space.

Commit:
CMSIS/CORTEX-M: Don't build mbed_tz_context.c in TF-M targets

- TF-M v8m secure-side implements their own TZ context APIs so need to avoid building the Mbed implementation.

(cherry picked from commit d3f7abdb7c109517e6a71daed8bae63ad6436afc)

Commit:
CMSIS/CORTEX-M: Don't build mbed_tz_context.c in TF-M targets

- TF-M v8m secure-side implements their own TZ context APIs so need to avoid building the Mbed implementation.

(cherry picked from commit d3f7abdb7c109517e6a71daed8bae63ad6436afc)

0xc0170 committed:
importer: fix byte like objects

0xc0170 committed:
importer: fix rtos paths

0xc0170 committed:
importer: remove rtos changed files

As they were applied on the different code base (moved files), we remove them and
reapply locally manually and add new sha.

0xc0170 committed:
[CMSIS_5]: Updated to b5f0603d6

0xc0170 committed:
cmsis importer: add sha for rtx changes after file moved

cherry pick otherwise fails because the sources were moved under rtos/source. I manually resolved these patches. For the future, lets use these newer sha that should work.

0xc0170 committed:
[CMSIS_5]: Updated to a65b7c9a3

0xc0170 committed:
importer: update readme

0xc0170 committed:
importer: add a note about cherry pick continue for conflicts

evedon commented on tools/importer/README.md:
```diff
@@ -3,7 +3,8 @@
 ### Pre-requisties
-1. Get the required repository clone and update it to commit required. Note: Repository should be placed outside the mbed-os.
+1. Get the required repository clone and update it to commit required - the recommended would be to checkout the latest release version in the repository that we are getting an update from.
+Note: Repository should be placed outside the mbed-os. For instance CMSIS_5 repo would be at the same level as mbed-os repository (`update/mbed-os` and update/CMSIS_5)
```
Could you add a link to the github repo for CMSIS_5?

0xc0170 committed:
importer: add CMSIS 5 repo link

evedon reviewed (dismissed):
Could you add in the title that his is the CMSIS 5.7.0 release

0xc0170 committed:
rtos: fix incorrect conflict resolution

We used previously __FPU_USED so stay with it in M33 startup file.

0xc0170 committed:
rtos: do not include RTE_Component header

RTE_Component can't be included in the assembly file, as it is application header file
(C/C++) therefore assembler fails in this case. If the symbol RTE_CMSIS_RTOS2_RTX5_ARMV8M_NS is
needed, just define it via config. Therefore I am leaving the rest of ifdef _RTE_ there.

0xc0170 committed:
tools: update importer sha

0xc0170 committed:
Use cmsis gcc types instead of own

This caused a conflict. As CMSIS update introduced low level init, lets use the types
from CMSIS. We could potentionally use __cmsis_start but as I saw for some targets,
the init routine is slightly different. So rather keep what we have in targets, and just
use types already defined in CMSIS.

0xc0170 committed:
MIMXRT1050: undefine vector table symbol from CMSIS (not used)

0xc0170 committed:
Revert "Use cmsis gcc types instead of own"

This reverts commit 606ccbceff0086f1b15969d9d7d40b415883a662.

0xc0170 committed:
Analog devices: overwrite cmsis startup

define __PROGRAM_START so we use own startup as AD had it defined, this fixes the conflicts with
CMSIS_5 update (they introduced low level init).

0xc0170 committed:
M251: fix startup after cmsis update

Don't use CMSIS low level startup

0xc0170 committed:
M2351: fix startup after cmsis update

Don't use CMSIS low level startup

0xc0170 committed:
TARGET_MIMXRT1050: fix vector table redef from cmsis

We need a linker script as it was. As files update in the future, should be compatible
with newer CMSIS core.

0xc0170 committed:
ad4050lz: fix gcc copy table and zero table missing

evedon reviewed (approved):
LGTM
